Automattic/wp-calypso
# Build SDK Gutenpack: Use file-loader to include images

#### Changes proposed in this Pull Request

* Add `.png`, `.jpg` and `.gif` files to the file loader config
* Add hash to file loader name
* Use the file loader to bundle Simple Payments block static assets
* Updates the jetpack-blocks scripts to
  - Add clean (removes build
  - Clean on publish

- [x] Requires a Jetpack change to include these assets: https://github.com/Automattic/jetpack/pull/10570

#### Testing instructions

- Ensure related posts placeholders continue to work well in WP Admin (Add Gutenberg related posts block)
- Ensure related posts placeholders continue to work well in Gutenberg wpcalypso (Add Gutenberg related posts block)
- Ensure related posts demo content continue to work well in Calypso (related posts settings for a Jetpack site) - https://calypso.live/settings/traffic/?branch=update/import-static-assets-gberg-simplepay
* Simple payments button is correctly bundled and loaded in wpcalypso
* Simple payments button is correctly bundled and loaded in Jetpack site WP Admin (https://jurassic.ninja/create?jetpack-beta&branch=update/gutenpack-include-full-build&shortlived&wp-debug-log&gutenberg&gutenpack&calypsobranch=update/import-static-assets-gberg-simplepay
)
* Do a publish dry run and verify produced files looks correct:
  ```sh
  cd client/gutenberg/extensions/presets/jetpack
  npm publish --dry-run
  ```

Something like this:

```
npm notice === Tarball Contents ===
npm notice 912B    package.json
npm notice 508B    README.md
npm notice 129B    build/block-manifest.json
npm notice 7.8kB   build/editor-beta.css
npm notice 178.9kB build/editor-beta.js
npm notice 7.8kB   build/editor-beta.rtl.css
npm notice 6.8kB   build/editor.css
npm notice 171.0kB build/editor.js
npm notice 6.8kB   build/editor.rtl.css
npm notice 7.5kB   build/images/paypal-button.png
npm notice 8.2kB   build/images/paypal-button@2x.png
npm notice === Tarball Details ===
npm notice name:          @automattic/jetpack-blocks
npm notice version:       6.8.0-beta.1
npm notice package size:  136.5 kB
npm notice unpacked size: 396.5 kB
npm notice shasum:        cd6e15374eec0316ebf74262f09e0f396b7e2fda
npm notice integrity:     sha512-x7bx+oSQ+2WTQ[...]RDlYSj80cxqjg==
npm notice total files:   11
npm notice
+ @automattic/jetpack-blocks@6.8.0-beta.1
```

Supersedes https://github.com/Automattic/wp-calypso/pull/28360 and https://github.com/Automattic/jetpack/pull/10559

Fixes #27838 (cc: @jeffersonrabb, please verify)

FYI: @Automattic/luna @Automattic/poseidon @Automattic/calypso-sdk
